kyo-core/shared/src/main/scala/kyo/AllowUnsafe.scala

```diff
@@ -0,0 +1,28 @@
+package kyo
+
+import kyo.Ansi.*
+import kyo.kernel.Safepoint
+import scala.annotation.implicitNotFound
+
+@implicitNotFound("""
+Unsafe operation detected!
+This code requires explicit permission to perform unsafe operations that
+break referential transparency.
+
+Recommended approach:
+Wrap your unsafe code in IO(...) to properly manage side effects:
+
+IO { /* your unsafe code here */ }
+
+For low-level integration or performance-sensitive code only:
+If you must proceed without IO, you have to explicitly embrace the danger:
+
+import AllowUnsafe.embrace.danger // Avoid!
+
+WARNING: Using AllowUnsafe directly bypasses important safety mechanisms.
+Ensure you fully understand the risks before proceeding this way.""")
+opaque type AllowUnsafe = Null
+
+object AllowUnsafe:
+    object embrace:
+        inline given danger: AllowUnsafe = null
```

Comment:
Is the idea that users would need to add `given AllowUnsafe = AllowUnsafe.embrace.danger` instead of `import AllowUnsafe.embrace.danger`? We'd still need a way to allow multiple unsafe calls in a scope without introducing `IO` suspensions. For example, a `Channel.Unsafe` impl will need to call multiple `Queue.Unsafe` methods without suspension and `IO` would be introduce by the `Channel(unsafe: UnsafeChannel)` wrapper.

Reply:
That's a valid concern but passing the evidence manually on each call feels too cumbersome if we consider unsafe access a first-class feature. Depending on where the unsafe code is used, it might not be trivial to pass the evidence as parameter as well. For example, it might not be possible to do that if we're implementing some interface required by an integration.

I feel the compile-time error message paired with the explicit indication of danger of the import is enough to avoid users overlooking the use of unsafe APIs. We can review the approach if there's indication of misuse due to convenience.